File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def sumNumbers(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: int
        """
        return self.dfs(root,0)
    def dfs(self,node,ans):
        if node == None:
            return 0
        ans = ans*10 + node.val
        if node.left == None and node.right == None:
            return ans
        return self.dfs(node.left,ans) + self.dfs(node.right,ans)
        

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return dfs(root,0);
    }
    public int dfs(TreeNode node, int ans){
        if(node == null) return 0;
        ans = ans*10 + node.val;
        if(node.left == null && node.right == null){
            return ans;
        }
        return dfs(node.left,ans) + dfs(node.right,ans);
    }
}